spec.md

```diff
@@ -477,7 +477,112 @@ class ClassName
 }
 ```
 
-### 4.4 Methods and Functions
+### 4.4 Property Hooks
+
+Object properties may also include hooks, which have a number of syntactic options.
+
+If there is only one hook implementation, and that hook implementation itself uses
+the short-hook syntax, then the hook declaration MAY be listed entirely inline.  The hook
+name MUST be separated from the opening brace and the arrow operator by a single
+space, and the semicolon ending of the hook MUST be separated from the cosing brace
+by a single space.  For example:
+
+```php
+class Example
+{
+    public string $myName { get => __CLASS__; }
+
+    public string $newName { set => ucfirst($value); }
+}
+```
+
+If those two criteria are not met, then the hook block MUST be spread across multiple lines:
+
+```php
+class Example
```

Comment:
This example does not match the preceding paragraph, because the criteria are still met.